crates/storage/db/src/tables/mod.rs

```diff
@@ -136,8 +129,8 @@ macro_rules! tables {
                 }
             }
 
-            impl $crate::table::Table for $name {
-                const TABLE: Tables = Tables::$name;
+            impl reth_db_api::table::Table for $name {
+                const NAME: &'static str = table_names::$name;
```

Comment:
Does this also address @prestwich's limitation of not being able to define custom tables in ExEx?

Reply:
not entirely, most of the code still assumes a fixed set of tables by using the generated `Tables` enum

Reply:
This LOC was one of the primary issues, but i haven't identified others yet

https://github.com/paradigmxyz/reth/pull/8594/files#diff-ee4d680a970f12705c9230a199a58e271d1f250dba479dea525f10c53294f1dfR80